src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,188 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
-  return (
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+import { PersonLink } from './PersonLink';
+import { SearchLink } from './SearchLink';
+import { useMemo } from 'react';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+  const gender = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const sortedOptions = ['Name', 'Sex', 'Born', 'Died'];
+
+  const filteredPeople = useMemo(() => {
+    let filtered = [...people];
+
+    if (query) {
+      const normalizedQuery = query.toLocaleLowerCase();
+
+      filtered = filtered.filter(
+        ({ name, motherName, fatherName }) =>
+          name.toLocaleLowerCase().includes(normalizedQuery) ||
+          motherName?.toLocaleLowerCase().includes(normalizedQuery) ||
+          fatherName?.toLocaleLowerCase().includes(normalizedQuery),
+      );
+    }
+
+    if (gender) {
+      filtered = filtered.filter(person => person.sex === gender);
+    }
+
+    if (centuries.length) {
+      filtered = filtered.filter(person =>
+        centuries.includes(
+          (+person.born.toString().slice(0, 2) + 1).toString(),
+        ),
+      );
+    }
+
+    return filtered;
+  }, [query, gender, centuries, people]);
+
+  const sortedPeople = useMemo(() => {
+    const sorted = [...filteredPeople].sort(
+      (personA: Person, personB: Person) => {
+        switch (sort) {
+          case 'name':
+          case 'sex':
+            return personA[sort].localeCompare(personB[sort]);
+          case 'born':
+          case 'died':
+            return personA[sort] - personB[sort];
+          default:
+            return 0;
+        }
+      },
+    );
+
+    if (order === 'desc') {
+      sorted.reverse();
+    }
+
+    return sorted;
+  }, [filteredPeople, sort, order]);
+
+  const toggleOrder = (sortName: string) => {
```

Comment:
The `toggleOrder` function is a bit confusing and could be simplified. It seems like you're trying to handle too many cases at once, which makes the function hard to understand and prone to bugs. Consider breaking it down into simpler, more manageable parts.